x/stakeibc/keeper/trade_route_test.go

```diff
@@ -96,3 +96,32 @@ func (s *KeeperTestSuite) TestGetAllTradeRoutes() {
 	actualRoutes := s.App.StakeibcKeeper.GetAllTradeRoutes(s.Ctx)
 	s.Require().ElementsMatch(expectedRoutes, actualRoutes)
 }
+
+func (s *KeeperTestSuite) TestGetTradeRouteFromTradeChainId() {
+	// Store 3 trade routes
+	for i := 1; i <= 3; i++ {
+		rewardDenom := fmt.Sprintf("reward-%d", i)
+		hostDenom := fmt.Sprintf("host-%d", i)
+		chainId := fmt.Sprintf("chain-%d", i)
+
+		s.App.StakeibcKeeper.SetTradeRoute(s.Ctx, types.TradeRoute{
+			RewardDenomOnRewardZone: rewardDenom,
+			HostDenomOnHostZone:     hostDenom,
+			TradeAccount: types.ICAAccount{
+				ChainId: chainId,
+			},
+		})
+	}
+
+	// Search for each of them by chain ID
+	for i := 1; i <= 3; i++ {
```

Comment:
also search for chainids that do _not_ have trade routes!

Reply:
good call! meant to add this and then totally forgot

Reply:
leaving open until resolved

Reply:
it was updated [here](https://github.com/Stride-Labs/stride/pull/1142/commits/31a7205b1038d1a40371d249a34537266ad518f1#diff-769192289f97a856eec27684fae4cf94c5d8a5990362349831ca7b01c4f7e61dR128-R130)